aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -694,6 +695,53 @@ Tensor new_empty_batching_rule(
   return physical_view.newLogicalFromPhysical(result);
 }
 
+Tensor new_empty_strided_batching_rule(
+    const Tensor& self,
+    IntArrayRef size,
+    IntArrayRef stride,
+    const TensorOptions& options) {
+  auto physical_view = MultiBatchVmapTransform::logicalToPhysical(self);
+  auto physical_size = physical_view.getPhysicalShape(size);
+
+  // Let [B0, B1, B2] be the shape of the batch dims. We know what the physical
+  // shape of the result should be ([B0, B1, B2] + size), but what about
+  // the physical strides?
+  //
+  // We're actually free to pick whatever stride we want:
+  // e.g., for size=[5, 3], stride=[0, 1], we could decide to
+  // use
+  // - physical size: [B0, B1, B2, 5, 3]
+  // - physical stride: [100*B1*B2, 100*B2, 100, 0, 1]
+  //
+  // Let's select the strides that result in the smallest memory usage.
```

Comment:
I am not sure about this argument. Even we we transposed everything to be "column contiguous" the actual memory usage would be the same no?
Is the choice here done to make the batch tensor contiguous if the original result of `empty_strided(size, stride)` was contiguous?

Reply:
If we picked `[100*B1*B2, 100*B2, 100, 0, 1]` instead of `[B1 * B2 *S, B2 * S, S, 5, 3]`, then the resulting storage size would be very large (100*B1*B2 * (B0-1) would be the memory location of result[B0-1], rather than B1 * B2 *S * (B0-1))

If the original result of `empty_strided(size, stride)` is contiguous, then the batched tensor is contiguous, yes -- that is also one of the motivations (I'll add that to the comment)

What do you mean by "column contiguous"?

Reply:
> What do you mean by "column contiguous"?

Sorry that is not very clear. Stride being strictly increasing. Matching a column-major packing.

> If we picked [100*B1*B2, 100*B2, 100, 0, 1] instead of [B1 * B2 *S, B2 * S, S, 5, 3]

You mean using another fixed values V (=100 here) such that S <= V?
I guess that is true but I'm not sure why would anyone argue that this is a thing we want to do?

Reply:
> You mean using another fixed values V (=100 here) such that S <= V?
I guess that is true but I'm not sure why would anyone argue that this is a thing we want to do?

Yes, exactly. It's a degree of freedom that I was shutting down for completeness. I'll refactor the comment to make it clear that the reason is more to make the batch tensor contiguous if the original result of empty_strided(size, stride) is contiguous (and if the original result is not contiguous, for the batch dimensions to be contiguous in memory)